include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp

```diff
@@ -136,12 +136,12 @@ inline alignment_t alignment_trace(database_t && database,
         {
             coordinate.col = std::max<size_t>(coordinate.col, 1) - 1;
             --end_aligned_db;
-            insert_gap(std::get<1>(aligned_seq), end_aligned_qy);
+            end_aligned_qy = insert_gap(std::get<1>(aligned_seq), end_aligned_qy);
```

Comment:
The return value of `insert_gap` is unfortunately not documented... can it possibly return a different value than the given `end_aligned_qy` parameter? 
If no, then this assignment is a no-op which is rather confusing than beneficial. 
If yes, why do we need to touch both db and qy variables while going left in the traceback? Also, then you might need to check if the returned value is still in the valid range.

Reply:
After @smehringer explained to me the meaning of the return value, I'm sure your change is good! Please add the `\returns` documentation for `insert_gap` instead.